src/HolographicCamera.Unity/Assets/HolographicCamera/Scripts/KeepDeviceActive.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+#if UNITY_WSA && WINDOWS_UWP
+    using System;
+    using Windows.System.Display;
+#endif
+
+public class KeepDeviceActive : MonoBehaviour
+{
+#if UNITY_WSA && WINDOWS_UWP
+    private static DisplayRequest displayRequest;
+#endif
+
+    private void Start()
+    {
+        CreateDisplayRequest();
+    }
+
+    private void OnDestroy()
+    {
+#if UNITY_WSA && WINDOWS_UWP
+        if (displayRequest != null)
+        {
+            try
+            {
+                displayRequest.RequestRelease();
+                Debug.Log("Display request released.");
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"Error releasing display request: {e.Message}");
+            }
+        }
+#endif
+    }
+
+    private void CreateDisplayRequest()
+    {
+#if UNITY_WSA && WINDOWS_UWP
+        UnityEngine.WSA.Application.InvokeOnUIThread(() =>
+        {
+            if (displayRequest == null)
+            {
+                try
+                {
+                    displayRequest = new DisplayRequest();
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError($"Error creating display request: {e.Message}");
+                }
+            }
+
+            if (displayRequest != null)
+            {
+                try
+                {
+                        // This call activates a display-required request. If successful,  
+                        // the screen is guaranteed not to turn off automatically due to user inactivity. 
+                        displayRequest.RequestActive();
```

Comment:
Thoughts on moving displayRequest.RequestActive() to the previous try block (i.e. right after creating it?)

I guess the downside is that you might not get specific Debug.LogError for creation vs requesting, but it seems a bit messy here to have a try/catch for each of these.

Reply:
I based this off of a sample, but reducing the length of this code seems like a good plan. I would hope that the exceptions would be different, I'm not really sure how to generate either of them.